Source/Gradient.swift

```diff
@@ -1,15 +1,54 @@
 import UIKit
 
-public struct GradientComponents {
+public protocol Gradient {
 
-    let color: UIColor
-    let location: Double
+    var angle: GradientAngle { get }
+    func makeGradientComponents() -> [GradientComponents]
 
 }
 
-public protocol Gradient {
+public struct GradientComponents {
+    
+    let color: UIColor
+    let location: Double
+    
+}
 
-    var startPoint: CGPoint { get }
-    func makeGradientComponents() -> [GradientComponents]
+public struct GradientPoints {
+    
+    public var start: CGPoint
```

Comment:
I'm thinking we might not need to expose this publicly, right? The developer constructing won't need to know how to construct the `GradientPoints`.

Reply:
It looks like we're only constructing these via an initializer, or accessing them. Should be safe to turn them into `let`s rather than `var`s.